iOS project export bound to Arkit, itself faulty

**Godot version:**
v3.2.beta.custom_build.500863859

**OS/device including version:**
macOS Mojave, iOS 13.1, 13.2 / iPhone SE, iPad Pro Models A1673 (1st gen) and A1980 (3rd gen)

**Issue description:**
1. iOS project export : when the Arkit Capabilities option is unchecked, the export archive fails with unresolved symbols:
```
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_ARSession", referenced from:
      objc-class-ref in Game.a(arkit_interface.iphone.debug.arm64.o)
  "_OBJC_CLASS_$_ARWorldTrackingConfiguration", referenced from:
      objc-class-ref in Game.a(arkit_interface.iphone.debug.arm64.o)
ld: symbol(s) not found for architecture arm64
```
This effectively foces Arkit inclusion in the project export to succeed, which is not wanted.

2. When successfully exporting the project with the Arkit Capabilities option checked and deploying the game on a device, the game asks for Camera access permission on startup. Refusing access proceeds as expected. However, granting camera access crashes the game with invalid thread access in CameraIOS::update_feeds(), line 362 (AVCaptureDeviceDiscoverySession *session = ...)

**Steps to reproduce:**

**Minimal reproduction project:**

--- a/modules/arkit/arkit_interface.mm
+++ b/modules/arkit/arkit_interface.mm
@@ -37,6 +37,8 @@
 #import <ARKit/ARKit.h>
 #import <UIKit/UIKit.h>
 
+#include <dlfcn.h>
+
 #include "arkit_interface.h"
 #include "arkit_session_delegate.h"
 
@@ -53,7 +55,10 @@ void ARKitInterface::start_session() {
 	// Ignore this if we're not initialized...
 	if (initialized) {
 		print_line("Starting ARKit session");
-		ARWorldTrackingConfiguration *configuration = [ARWorldTrackingConfiguration new];
+
+		Class ARWorldTrackingConfigurationClass = NSClassFromString(@"ARWorldTrackingConfiguration");
+		ARWorldTrackingConfiguration *configuration = [ARWorldTrackingConfigurationClass new];
+
 		configuration.lightEstimationEnabled = light_estimation_is_enabled;
 		if (plane_detection_is_enabled) {
 			configuration.planeDetection = ARPlaneDetectionVertical | ARPlaneDetectionHorizontal;
@@ -221,7 +226,17 @@ bool ARKitInterface::initialize() {
 		print_line("initializing ARKit");
 
 		// create our ar session and delegate
-		ar_session = [ARSession new];
+		Class ARSessionClass = NSClassFromString(@"ARSession");
+		if (ARSessionClass == Nil) {
+			void *arkit_handle = dlopen("/System/Library/Frameworks/ARKit.framework/ARKit", RTLD_NOW);
+			if (arkit_handle) {
+				ARSessionClass = NSClassFromString(@"ARSession");
+			} else {
+				print_line("ARKit init failed");
+				return false;
+			}
+		}
+		ar_session = [ARSessionClass new];
 		ar_delegate = [ARKitSessionDelegate new];
 		ar_delegate.arkit_interface = this;
 		ar_session.delegate = ar_delegate;

